commands/variable/set/set.go

```diff
@@ -0,0 +1,170 @@
+package set
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/profclems/glab/commands/cmdutils"
+	"github.com/profclems/glab/internal/glrepo"
+	"github.com/profclems/glab/internal/utils"
+	"github.com/profclems/glab/pkg/api"
+	"github.com/spf13/cobra"
+	"github.com/xanzy/go-gitlab"
+)
+
+type SetOpts struct {
+	HTTPClient func() (*gitlab.Client, error)
+	IO         *utils.IOStreams
+	BaseRepo   func() (glrepo.Interface, error)
+
+	Key       string
+	Value     string
+	Type      string
+	Scope     string
+	Protected bool
+	Masked    bool
+	Group     string
+}
+
+func NewVariableCmd(f *cmdutils.Factory, runE func(opts *SetOpts) error) *cobra.Command {
+	opts := &SetOpts{
+		IO: f.IO,
+	}
+
+	validKeyMsg := "A valid key must have no more than 255 characters; only A-Z, a-z, 0-9, and _ are allowed"
+
+	cmd := &cobra.Command{
+		Use:     "set <key> <value>",
+		Short:   "Create a new project or group variable",
+		Aliases: []string{"new", "create"},
+		Args:    cobra.RangeArgs(1, 2),
+		Example: heredoc.Doc(`
+			$ glab variable set WITH_ARG "some value"
+			$ glab variable set FROM_FLAG -v "some value"
+			$ glab variable set FROM_ENV_WITH_ARG "${ENV_VAR}"
+			$ glab variable set FROM_ENV_WITH_FLAG -v"${ENV_VAR}"
+			$ glab variable set FROM_FILE < secret.txt
+			$ cat file.txt | glab variable set SERVER_TOKEN
+			$ cat token.txt | glab variable set GROUP_TOKEN -g mygroup --scope=prod
```

Comment:
Is it possible to use redirection `<`.

something like this: `glab variable set SERVER_TOKEN < file.txt`

Reply:
Yes I added an example. Check the examples